### ISSUE TITLE
update ldclient-js version

### ISSUE THREAD

**Issue by hartzis** [TrueCar/react-launch-darkly#81]
thoughts on bumping the `ldclient-js` so we get all the bug fixes and improvements?

It is currently at [2.7.1](https://github.com/launchdarkly/js-client/releases).

The [major changes in 2.0.0](https://github.com/launchdarkly/js-client/releases/tag/2.0.0) don't look too upsetting.

**Comment by sethbattin** [comment]
Certainly.  Another major version release for us.

I'd be curious if we could apply a peer dependency here, so that we don't have to care which version is available.  Those release notes seem centered on analytics, which our lib doesn't care about (I'm 90% certain about that, anyway).  So if some heavy user of LD wants our lib, but doesn't want to edit their own usage...maybe that matters.

Though I'd be equally surprised if someone was hopelessly suck on an old version of the client, but the _must_ upgrade this lib.  We already cut off react below 16.3, so it's a short lifetime at best.

Yep, that's all the concern I can muster.  By all means submit a PR, and thank you in advance!

**Comment by hartzis** [comment]
Nice, yeah was thinking peer dep too. and/or maybe a way to supply the LD client object to `react-launch-darkly` so the `ldclient-js` could be dynamically loaded client side too?

**Comment by sethbattin** [comment]
Dynamic loading sounds a little risky in terms of performance.  Adding extra ms ahead of flag rendering could be an unwanted effect.

I waffled a lot above, but i think an upgrade without extra complexity would be fine.  Our usage numbers are not yet so huge that we have to cater to perfect backwards compatibility.  If it ever grows, we can release _another_ major version bump to become ambidextrous.  In the meantime, if someone has very specific upgrade limitations, I would probably tell them to simply fork the library and overwrite it.

**Comment by hartzis** [comment]
let me just say what ya'll have created here is amazing! Does a PR to introduce `ldclient-js` as a peer dep with `*` or `^1.1.12 || ^2.0.0`?

I am looking to use this, or at least base our implementation off of it. We're using SSR and would be hydrating the flags with the `ldclient-node` and that's why I think we'd like to dynamically load the `ldclient-js` since the flags would already be rendered appropriately from the server.

Though it's bundle size is only estimate at `7.4kb` minified+gzip - https://bundlephobia.com/result?p=ldclient-js@2.7.1 🤔 maybe that isn't worth it.

**Comment by hartzis** [comment]
@jacobmoretti thank you for the awesome feedback. For you SSR and hydration approach are you using `allFlagsState` in the node-sdk: https://docs.launchdarkly.com/v2.0/docs/node-sdk-reference#section-all-flags ? or are you somehow calling `variation` only when SSR'ing a page/route that has a flag needed?

**Comment by jacobmoretti** [comment]
@hartzis Yes, we're using the `allFlagsState` from their server client. Only difference is we're using LaunchDarkly's ruby client. Typically in our app, we only allow SSR of feature flags where absolutely needed. Mostly so we can continue to cache where applicable.
